### PR TITLE
fix: Allow fetchMore() to be used with 'no-cache' fetchPolicy

### LINKED ISSUE #5239: Results of `updateQuery` ignored for `fetchPolicy: no-cache`
**Intended outcome:**
For a Query with `fetchPolicy: no-cache`
When I call `fetchMore`
And I provide an `updateQuery` function to merge in new results
I expect the Query to re-render
With the new results available in `data`

**Actual outcome:**
However the component does not re-render with new results.

If I remove the `fetchPolicy: no-cache`,
The component re-renders with new data as expected when `fetchMore` is called.

**How to reproduce the issue:**
https://codesandbox.io/embed/apollo-client-error-template-gfc7l

With `fetchPolicy: cache-first`, clicking "Fetch more" adds "Another Person" to the list.

With `fetchPolicy: no-cache`, clicking "Fetch more" does nothing.

**Versions**
The above codesandbox.io includes its own list of versions.

But the issue also shows up in our app code on my local machine, which has the following versions:

```
  System:
    OS: macOS 10.14.5
  Binaries:
    Node: 8.11.3 - ~/.nvm/versions/node/v8.11.3/bin/node
    Yarn: 1.17.3 - ~/Projects/wonderschool/mobile/node_modules/.bin/yarn
    npm: 5.6.0 - ~/.nvm/versions/node/v8.11.3/bin/npm
  Browsers:
    Chrome: 76.0.3809.100
    Firefox: 66.0.3
    Safari: 12.1.1
  npmPackages:
    apollo: ^2.16.0 => 2.16.0 
    apollo-boost: ^0.4.4 => 0.4.4 
    react-apollo: ^3.0.1 => 3.0.1 
```

### PR DESCRIPTION
**Pull request by mogelbrod**
Happy to add test(s) if someone could verify that I've provided a valid fix for the issue.

This PR is basically a continuation of https://github.com/apollographql/apollo-client/issues/5239#issuecomment-595248035

Fixes #5239

### Checklist:

- [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [ ] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

File: src/core/ObservableQuery.ts [hunk 1]
```diff
@@ -310,7 +311,7 @@ export class ObservableQuery<
           ...fetchMoreOptions.variables,
         },
       }),
-      fetchPolicy: 'network-only',
+      fetchPolicy: fetchPolicy === 'no-cache' ? 'no-cache' : 'network-only',
```
Comment (mogelbrod) [inline src/core/ObservableQuery.ts]: WIthout this `markQueryResult` appears to write to the cache:

https://github.com/apollographql/apollo-client/blob/e9be4bea5d34ff73c04355ba2cd248ced779eddc/src/core/QueryManager.ts#L505-L519

File: src/core/ObservableQuery.ts [hunk 2]
```diff
@@ -331,7 +332,7 @@ export class ObservableQuery<
               fetchMoreResult: data,
               variables: combinedOptions.variables as TVariables,
             }) : data;
-          });
+          }, fetchPolicy === 'no-cache');
```
Comment (mogelbrod) [inline src/core/ObservableQuery.ts]: This argument might be better off as an object?

File: src/core/QueryManager.ts [hunk 3]
```diff
@@ -1339,7 +1339,7 @@ export class QueryManager<TStore> {
     );
   }
 
-  private setQuery<T extends keyof QueryInfo>(
+  public setQuery<T extends keyof QueryInfo>(
```
Comment (mogelbrod) [inline src/core/QueryManager.ts]: Required for access from `ObservableQuery.updateQuery`

**Comment by mikedizon** [comment]
Waiting for this!

**Comment by mogelbrod** [comment]
Would love to get this into the v3.0 release, just waiting for someone knowledgeable (@hwillson perhaps?) of the code base to give some feedback on the approach I used before writing the tests for it.

**Comment by mirague** [comment]
Would love to hear from the team on this!

**Comment by korengrip** [comment]
@mogelbrod any news?

**Comment by mogelbrod** [comment]
> @mogelbrod any news?
> Really need this feature.

We haven't started our migration to Apollo yet, so I haven't been able to check if [this merged v3 PR](https://github.com/apollographql/apollo-client/pull/6221#issuecomment-625201470) solved this issue. Hopefully it did though.

**Comment by korengrip** [comment]
> > @mogelbrod any news?
> > Really need this feature.
> 
> We haven't started our migration to Apollo yet, so I haven't been able to check if [this merged v3 PR](https://github.com/apollographql/apollo-client/pull/6221#issuecomment-625201470) solved this issue. Hopefully it did though.

It seems like it didn't solve the issue.

**Comment by hwillson** [comment]
Thanks for this @mogelbrod and sorry for the delay. This work was superseded by the work in https://github.com/apollographql/apollo-client/pull/6221, which set the `fetchMore` `fetchPolicy` default to `no-cache` (see the description in that PR for the reasoning):

https://github.com/apollographql/apollo-client/blob/7041fe744233d91813368beb73f6993f86a50189/src/core/ObservableQuery.ts#L413-L418

Thanks!

**Pull request closed without merging**